Nuget Packaging is broken with Xamarin.iOS 15.2.x.x

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->
### Steps to Reproduce

#### IDE
1. use VisualStudio 2019 with Xamarin.iOS 15.2.0.1 or 15.2.0.17
2. create a Xamarin.iOS project
3. run the nuget pack command

#### CLI
or run `msbuild x.csproj -t:pack`

### Expected Behavior

a new nuget gets created as with previous versions

### Actual Behavior

the nuget packaging stops with the following error:

```
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Shared.targets(232,3): error MSB4036: The "GetNuGetShortFolderName" task was not found. Check the following: 1.) The name of the task in the project file is the same as the name of the task class. 2.) The task class is "public" and implements the Microsoft.Build.Framework.ITask interface. 3.) The task is correctly declared with <UsingTask> in the project file, or in the *.tasks files located in the "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin" directory.
```
### Environment

```
Visual Studio Professional 2019 for Mac
Version 8.10.13 (build 11)
Installation UUID: 5ea74647-5965-4d67-aa28-462a22a4602f
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

Mono Framework MDK
Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

Roslyn (Language Service)
3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

NuGet
Version: 5.9.0.7134

.NET Core SDK
SDK: /usr/local/share/dotnet/sdk/5.0.401/Sdks
SDK Versions:
	5.0.401
	3.1.413
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

.NET Core Runtime
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.12
	5.0.10
	3.1.19
	3.1.10
	2.1.23

.NET Core 3.1 SDK
SDK: 3.1.413

Xamarin.Profiler
Version: 1.6.12.29
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin Designer
Version: 16.11.0.39
Hash: cd672761d
Branch: remotes/origin/d16-11
Build date: 2021-10-06 20:09:18 UTC

Xamarin.Android
Version: 12.0.0.3 (Visual Studio Professional)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/Shared/Developer/android/sdk
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 31.0.3
SDK Build Tools Version: 32.0.0 rc1

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/jdk-11.0.3.jdk/Contents/Home
2019-04-16
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:12 UTC

Android Device Manager
Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:30 UTC

Apple Developer Tools
Xcode 13.1 (19466)
Build 13A1030d

Xamarin.Mac
Version: 8.2.0.1 (Visual Studio Professional)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:55-0400

Xamarin.iOS
Version: 15.2.0.1 (Visual Studio Professional)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:56-0400

Build Information
Release ID: 810130011
Git revision: 53e3946a0f6b630a1124eb7ff26c1b545d040d32
Build date: 2021-11-04 11:20:46-04
Build branch: release-8.10

Operating System
Mac OS X 10.16.0
Darwin 21.1.0 Darwin Kernel Version 21.1.0
    Wed Oct 13 17:33:23 PDT 2021
    root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64
```
### Build Logs

<!--
1. Place cursor below this comment block.
2. Attach build log or link to gist (https://gist.github.com/) of the log.
3. Logs can be found in the Build Output tab of the Errors pad in Visual Studio
To get full build logs, set the log verbosity to diagnostic at the following locations:
- On Visual Studio for Mac: Preferences > Projects > Build
- On Visual Studio for Windows: Tools > Options > Projects and Solutions > Build and Run
On Visual Studio for Windows you also want to add `-v -v -v -v` to the mtouch additional arguments by right-clicking the project in the solution explorer and selecting `Properties`.
-->
```
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Shared.targets(232,3): error MSB4036: The "GetNuGetShortFolderName" task was not found. Check the following: 1.) The name of the task in the project file is the same as the name of the task class. 2.) The task class is "public" and implements the Microsoft.Build.Framework.ITask interface. 3.) The task is correctly declared with <UsingTask> in the project file, or in the *.tasks files located in the "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin" directory.
```


### Example Project (If Possible)

<!--
1. Place cursor below this comment block.
2. Drag and drop the compressed project or files needed to reproduce.
-->

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

@sbwrk  Thanks for the clarification.

We did now also create an example project and used the following command to build this on our environment:

`msbuild Issue13469.csproj -t:pack`

as expected we got the following result:

```
msbuild Issue13469.csproj -t:pack
Microsoft (R) Build Engine version 16.9.0 for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 12/1/2021 10:11:03 AM.
Project "/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" on node 1 (pack target(s)).
_DetectDebugNetworkConfiguration:
    DebugIPAddresses: <null>
_BeforeCoreCompileImageAssets:
Skipping target "_BeforeCoreCompileImageAssets" because all output files are up-to-date with respect to the input files.
_CoreCompileImageAssets:
Skipping target "_CoreCompileImageAssets" because all output files are up-to-date with respect to the input files.
_BeforeCoreCompileInterfaceDefinitions:
Skipping target "_BeforeCoreCompileInterfaceDefinitions" because it has no inputs.
_BeforeCoreCompileInterfaceDefinitions:
Skipping target "_BeforeCoreCompileInterfaceDefinitions" because it has no inputs.
_BeforeCoreCompileInterfaceDefinitions:
Skipping target "_BeforeCoreCompileInterfaceDefinitions" because it has no inputs.
_CoreCompileInterfaceDefinitions:
Skipping target "_CoreCompileInterfaceDefinitions" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_BeforeCoreCompileSceneKitAssets:
Skipping target "_BeforeCoreCompileSceneKitAssets" because it has no inputs.
_CoreCompileSceneKitAssets:
Skipping target "_CoreCompileSceneKitAssets" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_BeforeCompileTextureAtlases:
Skipping target "_BeforeCompileTextureAtlases" because it has no inputs.
_CoreCompileTextureAtlases:
Skipping target "_CoreCompileTextureAtlases" because it has no inputs.
_BeforeCompileCoreMLModels:
Skipping target "_BeforeCompileCoreMLModels" because all output files are up-to-date with respect to the input files.
_CoreCompileCoreMLModels:
Skipping target "_CoreCompileCoreMLModels" because all output files are up-to-date with respect to the input files.
_CoreOptimizePngImages:
Skipping target "_CoreOptimizePngImages" because it has no outputs.
_CoreOptimizePropertyLists:
Skipping target "_CoreOptimizePropertyLists" because it has no inputs.
_CoreOptimizeLocalizationFiles:
Skipping target "_CoreOptimizeLocalizationFiles" because it has no inputs.
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
  /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Roslyn/csc.exe /noconfig /fullpaths /nostdlib+ /errorreport:prompt /warn:4 /define:__IOS__;__MOBILE__;__UNIFIED__;DEBUG /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/mscorlib.dll /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.Core.dll /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.dll /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/Facades/System.Drawing.Common.dll /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.Xml.dll /reference:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/Xamarin.iOS.dll /debug+ /debug:portable /optimize- /out:obj/Debug/Issue13469.dll /target:library /utf8output /langversion:7.3 Class1.cs Properties/AssemblyInfo.cs "obj/Debug/Xamarin.iOS,Version=v1.0.AssemblyAttributes.cs"
  Using shared compilation with compiler from directory: /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/Roslyn
CopyFilesToOutputDirectory:
  Copying file from "/Users/John/Projects/Issue13469/Issue13469/obj/Debug/Issue13469.dll" to "/Users/John/Projects/Issue13469/Issue13469/bin/Debug/Issue13469.dll".
  Issue13469 -> /Users/John/Projects/Issue13469/Issue13469/bin/Debug/Issue13469.dll
  Copying file from "/Users/John/Projects/Issue13469/Issue13469/obj/Debug/Issue13469.pdb" to "/Users/John/Projects/Issue13469/Issue13469/bin/Debug/Issue13469.pdb".
IncrementalClean:
  Deleting file "/Users/John/Projects/Issue13469/Issue13469/obj/Debug/Issue13469.csproj.AssemblyReference.cache".
Project "/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (1) is building "/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (1:3) on node 1 (_GetTfmSpecificContentForPackage target(s)).
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Shared.targets(238,3): error MSB4036: The "GetNuGetShortFolderName" task was not found. Check the following: 1.) The name of the task in the project file is the same as the name of the task class. 2.) The task class is "public" and implements the Microsoft.Build.Framework.ITask interface. 3.) The task is correctly declared with <UsingTask> in the project file, or in the *.tasks files located in the "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin" directory. [/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj]
Done Building Project "/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (_GetTfmSpecificContentForPackage target(s)) -- FAILED.
Done Building Project "/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (pack target(s)) -- FAILED.

Build FAILED.

"/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (pack target) (1) ->
"/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj" (_GetTfmSpecificContentForPackage target) (1:3) ->
(_IncludeBindingResourcesInNuGetPackage target) ->
  /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Shared.targets(238,3): error MSB4036: The "GetNuGetShortFolderName" task was not found. Check the following: 1.) The name of the task in the project file is the same as the name of the task class. 2.) The task class is "public" and implements the Microsoft.Build.Framework.ITask interface. 3.) The task is correctly declared with <UsingTask> in the project file, or in the *.tasks files located in the "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin" directory. [/Users/John/Projects/Issue13469/Issue13469/Issue13469.csproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.04
```

See also the attached project and binary build log.

[Issue13469.binlog.zip](https://github.com/xamarin/xamarin-macios/files/7632492/Issue13469.binlog.zip)
[Issue13469.zip](https://github.com/xamarin/xamarin-macios/files/7632468/Issue13469.zip)


Thanks for your support.

--- a/msbuild/Xamarin.Shared/Xamarin.Shared.targets
+++ b/msbuild/Xamarin.Shared/Xamarin.Shared.targets
@@ -240,10 +240,20 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 		</PropertyGroup>
 		<Error Condition="'$(_HasOldStyleBindingItems)' == 'true'" Text="Creating a NuGet package is not supported for projects that have ObjcBindingNativeLibrary items. Migrate to use NativeReference items instead." />
 
+		<!-- The 'GetNuGetShortFolderName' task only exists for .NET, so hardcode values for _NuGetShortFolderName for legacy builds -->
+		<PropertyGroup Condition="'$(UsingAppleNETSdk)' != 'true'">
+			<_NuGetShortFolderName Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.iOS'">xamarinios10</_NuGetShortFolderName>
+			<_NuGetShortFolderName Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.TVOS'">xamarintvos10</_NuGetShortFolderName>
+			<_NuGetShortFolderName Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.WatchOS'">xamarinwatch10</_NuGetShortFolderName>
+			<_NuGetShortFolderName Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.Mac'">xamarinmac10</_NuGetShortFolderName>
+		</PropertyGroup>
+
 		<!-- Figure out where to place the binding resources (see references above for more info) -->
 		<GetNuGetShortFolderName
 			TargetFrameworkMoniker="$(TargetFrameworkMoniker)"
-			TargetPlatformMoniker="$(TargetPlatformMoniker)">
+			TargetPlatformMoniker="$(TargetPlatformMoniker)"
+			Condition="'$(_NuGetShortFolderName)' == ''"
+			>
 			<Output TaskParameter="NuGetShortFolderName" PropertyName="_NuGetShortFolderName" />
 		</GetNuGetShortFolderName>
 

